horqrux/differentiation/utils.py

```diff
@@ -0,0 +1,7 @@
+from __future__ import annotations
+
+from horqrux.primitives import Parametric, Primitive
+
+
+def is_parametric(gate: Primitive) -> bool:
```

Comment:
I was wondering if it was best for this predicate to stay as a standalone function or become a method in `Primitive`. I have a slight preference for the later but it's up to you.